x-pack/filebeat/input/entityanalytics/internal/collections/uuid_set.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package collections
+
+import (
+	"bytes"
+	"encoding/json"
+	"sort"
+
+	"github.com/google/uuid"
+)
+
+type UUIDSet struct {
```

Comment:
It's needed for the UnmarshalJSON method, since it requires a pointer receiver. Other methods for the definition you mentioned wouldn't make sense to use a pointer receiver, so then I'd be mixing value and pointer receivers, which is discouraged by Go and would likely trip the linter. I actually tried using `type UUIDSet map[uuid.UUID]struct{}` but couldn't figure out a way to get it to work.

As of right now, I don't think there's a requirement for it being thread-safe, but it will be trivial to add it later if needed.

In case you're wondering why I have a custom UnmarshalJSON, I marshal `UUIDSet` (backed by a map) to a list in JSON. A dictionary structure is too verbose for what I'm trying to represent here.